### PR TITLE
Implement blend options for grid

### PR DESCRIPTION
**Pull request by 0x00002a**
## What kind of change does this PR introduce?
- Feature

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- Ish, removes the effect of `g:neovide_floating_opacity` at least for the grid renderer

This implements usage of the `blend` option in the style in `grid_renderer`. It _seems_ to match up with `winblend` and obey local vs global options but actually testing it is proving to be quite difficult, and I don't fully understand how the drawing actually works (`RenderedWindow::draw` doesn't seem to actually do anything with its opacity for example).

So far it seems to work correctly for `nnn` in picker mode, but telescope is acting weird: when neovim is doing the drawing it kind of works in that the search box respects the transparency but nothing else does, in multigrid mode it just seems to ignore it completely.

I'm making this a draft because I am not confident that this actually works properly yet. My main concerns are:
- what is `RenderedWindow::draw` doing with its alpha, changing it doesn't seem to have any effect
- I don't really understand how the blend option is getting updated or how it syncs, so I'm not sure if it actually covers everything

Edit: OK I may have done a dumb, telescope has its own config setting for winblend :woman_facepalming: setting that works, and as far as I can tell it stays local to the window

### CODE REVIEW

**Comment by 0x00002a** [comment]
I've used this now for 5 months, it seems to work. Still not 100% sure on _why_ it works but I'll leave that to anyone who wants to review it

**Comment by 0x00002a** [comment]
> However the solution in this PR requires the user to change the highlight groups of what needs to be transparent

@LoipesMas Weirdly enough `pumblend` and `winblend` both seem to take effect here, which is what I was trying to say was confusing me in the original comment. I'm not sure why they would given like you say this is for the highlight group but :woman_shrugging: 

This is me changing pumblend on my local version:
https://user-images.githubusercontent.com/46329225/178002272-ec6689fe-725b-4509-b901-0cdc802ec809.mp4

File: src/renderer/grid_renderer.rs [hunk 1]
```diff
@@ -108,7 +108,7 @@ impl GridRenderer {
 
         if is_floating {
             self.paint
-                .set_alpha((255.0 * SETTINGS.get::<RendererSettings>().floating_opacity) as u8);
+               .set_alpha((255.0 * ((100 - style.blend) as f32 / 100.0)) as u8);
```
[the remark this answers is not in the document]
Comment (0x00002a) [reply src/renderer/grid_renderer.rs:111]: I've opened #1392 for this bug

**Comment by 0x00002a** [comment]
I've added clamps on when the blend is initially got from neovim and also added a fix for #1392 since it came up while testing it. I can break the fix out into another PR if you'd prefer

**Comment by MultisampledNight** [comment]
It's fine, no worries. I'm not sure whether this might be a thing for upstream though.

**Comment by LoipesMas** [comment]
I think it makes sense to also file this bug on neovim's github (and link this issue).

**Comment by 0x00002a** [comment]
I've reported it here: https://github.com/neovim/neovim/issues/19340

**Comment by 0x00002a** [comment]
Anything still blocking this merging?

**Comment by MultisampledNight** [comment]
Nope. Sorry for losing track on this, I'll merge this now.

**Comment by MultisampledNight** [comment]
(I'm quite confident that a whitespace change executed by `cargo fmt` won't change the CI state, so I'll just yolo this) Thank you!

**Pull request merged**